src/dynamic/LI_generator.cpp

```diff
@@ -79,7 +70,7 @@ std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> create_LI_model(uint
             //Last level does not have atomics
             for(int idx_atomic=0; idx_atomic < width-1; idx_atomic++) {
                 std::string atomic_name = "devstone_atomic_L" + std::to_string(level) + "_" + std::to_string(idx_atomic);
-                atomics_current_level.push_back(cadmium::dynamic::translate::make_dynamic_atomic_model<configured_atomic_devstone, TIME>(atomic_name));
+                atomics_current_level.push_back(make_atomic_devstone(atomic_name));
```

Comment:
You can pass the size as hint to the vector for allocation on construction to avoid it growing many times. You know the size of it in advance.